### PR TITLE
feat(core): add doc info modal

### PR DESCRIPTION
**Pull request by JimmFly**
close AF-1038
close AF-1039
close AF-1040
close AF-1046

A popup window has been added to facilitate viewing of this doc's info in edgeless mode and other modes.

https://github.com/toeverything/AFFiNE/assets/102217452/d7f94cb6-7e32-4ce7-8ff4-8aba1309b331

### CODE REVIEW

**Comment by JimmFly** [comment]
* **#7409** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7409?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @JimmFly and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/frontend/core/src/components/affine/page-properties/info-modal/time-row.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+                    },
+                  })
+                : null
+            }
+          />
+        ) : manager.updatedDate ? (
+          <RowComponent
+            icon={<HistoryIcon />}
+            name={t['Updated']()}
+            time={i18nTime(manager.updatedDate)}
```
Comment (pengx17) [inline packages/frontend/core/src/components/affine/page-properties/info-modal/time-row.tsx]: It seems updatedDate & createDate should use the same date format to be more consistent.

File: packages/frontend/core/src/components/blocksuite/block-suite-header/menu/index.tsx [hunk 2]
```diff
@@ -83,6 +87,14 @@ export const PageHeaderMenuButton = ({
     return setOpenHistoryTipsModal(true);
   }, [setOpenHistoryTipsModal, workspace.flavour]);
 
+  const setOpenInfoModal = useSetAtom(openInfoModalAtom);
+  const openInfoModal = () => {
+    mixpanel.track('Button', {
+      resolve: 'OpenDocInfoModal',
+    });
```
Comment (pengx17) [inline packages/frontend/core/src/components/blocksuite/block-suite-header/menu/index.tsx]: please confirm the payload with the product team

File: tests/affine-local/e2e/doc-info-modal.spec.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  clickNewPageButton,
+  clickPageMoreActions,
+  getBlockSuiteEditorTitle,
+  getPageByTitle,
+  getPageOperationButton,
+  waitForEditorLoad,
+} from '@affine-test/kit/utils/page-logic';
+import { expect } from '@playwright/test';
+
+test('New a page and open it ,then open info modal in the title bar', async ({
```
Comment (pengx17) [inline tests/affine-local/e2e/doc-info-modal.spec.ts]: add a test case to make sure forms in the modal is editable.

File: packages/frontend/core/src/components/blocksuite/block-suite-header/menu/index.tsx (hunk 2 above)
Comment (JimmFly) [reply packages/frontend/core/src/components/blocksuite/block-suite-header/menu/index.tsx]: I will remove them first, and wait for the discussion to create another PR about ~~buried points~~ event tracking.

Comment (pengx17) [reply packages/frontend/core/src/components/blocksuite/block-suite-header/menu/index.tsx]: "buried points"

File: packages/frontend/core/src/components/affine/page-properties/info-modal/info-modal.tsx [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+    onOpenChange(false);
+  }, [onOpenChange]);
+
+  const docsSearchService = useService(DocsSearchService);
+  const references = useLiveData(
+    useMemo(
+      () => LiveData.from(docsSearchService.watchRefsFrom(page.id), null),
+      [docsSearchService, page.id]
+    )
+  );
```
Comment (forehalo) [inline packages/frontend/core/src/components/affine/page-properties/info-modal/info-modal.tsx:53]: this is insane @EYHN

File: tools/cli/src/webpack/runtime-config.ts [hunk 5]
```diff
@@ -122,6 +124,11 @@ export function getRuntimeConfig(buildFlags: BuildFlags): RuntimeConfig {
       : buildFlags.mode === 'development'
         ? true
         : currentBuildPreset.allowLocalWorkspace,
+    enableInfoModal: process.env.ENABLE_INFO_MODAL
```
Comment (forehalo) [inline tools/cli/src/webpack/runtime-config.ts]: there is no need to add an env for it. it's not gonna be a optional feature after release

**Review by forehalo — APPROVED** [review]

**Pull request merged**